net-misc/x11-ssh-askpass/x11-ssh-askpass-1.2.4.1-r3.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs
+
+DESCRIPTION="X11-based passphrase dialog for use with OpenSSH"
+HOMEPAGE="https://github.com/sigmavirus24/x11-ssh-askpass"
+SRC_URI="https://github.com/sigmavirus24/${PN}/archive/${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+LICENSE="HPND"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE=""
+
+RDEPEND="virtual/ssh
+	x11-libs/libICE
+	x11-libs/libSM
+	x11-libs/libX11
+	x11-libs/libXt"
+DEPEND="${RDEPEND}"
+BDEPEND="app-text/rman
+	x11-misc/imake"
+
+src_prepare() {
+	default
+
+	# wrt #725200
+	printf '#!/bin/sh\n%s ${*}\n' "$(tc-getLD)" > "${T}"/ld \
+		|| die "printf failed"
+	chmod +x "${T}"/ld || die "chmod failed"
+	export PATH="${T}:${PATH}" || die "export failed"
```

Comment:
Yeeaahh, this seems weird. This is usually solvable with a single `sed` line, but rather just patch it then if not.

Reply:
I've seen this in a few ebuilds and it needs to stop (don't copy these).

If user happens to set LD=ld you have an infinite self-call, like [bug 764506](https://bugs.gentoo.org/764506) with CC=gcc

That aside, you also shouldn't fix this at all. The problem is with xmkmf, not this package. See [bug 729630#c7](https://bugs.gentoo.org/729630#c7)

Reply:
> That aside, you also shouldn't fix this at all. The problem is with xmkmf, not this package.

I might add that while it'll fail trying to find ld/gcc during xmkmf (if -native-symlinks), during building it's actually using the proper tools. I can even build this with clang and ld.lld